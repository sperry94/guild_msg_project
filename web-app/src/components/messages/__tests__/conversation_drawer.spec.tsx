import React from "react"
import renderer from "react-test-renderer"
import ConversationDrawer, {
  ConversationDrawerProps,
} from "../conversation_drawer"

jest.mock("../../../services/auth")
jest.mock("../../../services/firebase")
jest.mock("../../../services/conversations")
jest.mock("../../../services/messages")

describe("ConversationDrawer", () => {
  const cases: ConversationDrawerProps[] = [
    {
      conversationId: undefined,
    },
    {
      conversationId: "convo-id",
    },
  ]

  cases.forEach((c, index) => {
    it(`renders for case #${index + 1}`, () => {
      const tree = renderer.create(<ConversationDrawer {...c} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
