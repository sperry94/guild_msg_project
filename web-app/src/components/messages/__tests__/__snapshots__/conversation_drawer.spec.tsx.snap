// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationDrawer renders for case #1 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-2"
>
  <div
    className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-3 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    style={
      Object {
        "visibility": undefined,
      }
    }
  >
    <div
      className="makeStyles-toolbarOffset-1"
    />
    <div
      className="MuiCircularProgress-root makeStyles-spinner-48 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-addNewBtn-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          New Conversation
        </p>
      </span>
    </button>
  </div>
</div>
`;

exports[`ConversationDrawer renders for case #2 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-2"
>
  <div
    className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-3 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    style={
      Object {
        "visibility": undefined,
      }
    }
  >
    <div
      className="makeStyles-toolbarOffset-1"
    />
    <div
      className="MuiCircularProgress-root makeStyles-spinner-48 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-addNewBtn-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          New Conversation
        </p>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
